pkg/cluster/provider/eksclusterprovider.go

```diff
@@ -495,12 +500,27 @@ func newEKSClusterManager(region string) (*eks.EKS, *autoscaling.AutoScaling, er
 		log.Warn().Msgf("Failed to load valid access key from secret. Err=%s", err)
 	}
 
+	// From gist: https://gist.github.com/stormcat24/e8172b4130776e486f2758508eb4f3aa
+	sess := session.Must(session.NewSession())
 	c := aws.NewConfig().
-		WithRegion(region).
-		WithCredentialsChainVerboseErrors(true)
+		WithCredentialsChainVerboseErrors(true).
+		WithCredentials(credentials.NewChainCredentials([]credentials.Provider{
+			&credentials.EnvProvider{},
+			&credentials.SharedCredentialsProvider{},
+			// Required for IRSA
+			stscreds.NewWebIdentityRoleProvider(
+				sts.New(sess),
```

Comment:
How are you getting the two environment variables into the running cluster controller image? 

From [README.md](https://github.com/kubecost/cluster-turndown/blob/develop/README.md?plain=1#L110) looks like the access key and id are provided via a kubectl secret as service-key.json. You should probably add `aws_role_arn` and `aws_web_identity_token_file` as new new variables isnt it?

Also genuine question , this would work for IAM role for service account, prior to this change it was tested for IAM users, does this change support IAM users without service account[IRSA] if not you probably want to do a string empty comparison on two new variables and create session the new way you want for when role and token are passed. 

Going to tag @michaelmdresser who has more experience here if my assessment is wrong here!